docs/ruleset_functions/get_energy_required_to_heat_swh_use.md

```diff
@@ -0,0 +1,71 @@
+- **RMD**
+
+Returns:
+- **energy_required**: A fixnum indicating the total btu required to heat the water indicated by the swh_use
+
+Function Call:
+
+- get_spaces_served_by_SWH_use
+- get_obj_by_id
+
+Data Lookup: None
+
+Logic:
+
+- get the schedule: `hourly_schedule = swh_use.use_multiplier_schedule`
+- get the temperature at fixture: `supply_temperature = swh_use.distribution_system.design_supply_temperature`
+- get the distribution system: `distribution_system = swh_use.served_by_distribution_system`
+- get the water used: `swh_use_value = swh_use.use`
+- get the inlet water temperature: `inlet_T_sched = distribution_system.entering_water_mains_temperature_schedule`
+- set the drain heat recovery efficiency to 0: `drain_heat_recovery_efficiency = 0`
+- get the drain heat recovery efficiency if heat is recovered by drain: `if(swh_use.is_heat_recovered_by_drain): drain_heat_recovery_efficiency = distribution_system.drain_heat_recovery_efficiency`
+- get the use_units: `use_units = swh_use.use_units`
+
+- now we need to get all of the spaces that the swh_use is applied to.  The rule is that if spaces reference the swh_use, then the use applies to only those spaces.  If the use references no spaces, it applies to all spaces in the building segment.
+- get the ids of spaces served by this swh_use: `space_ids = get_spaces_served_by_SWH_use(RMD, swh_use)`
+- convert the space ids to spaces: `spaces = [get_obj_by_id(space_id, RMD) for space_id in space_ids]`
+- convert the swh_water_used from whatever units it's currently in to volume (assumed to be gallons) OR, if the use units are one of the power types, calculate the energy required directly
+- if the use_units is POWER_PER_PERSON: `if use_units == "POWER_PER_PERSON":`
+  - set power equal to the swh_use_value * sum of the occupants in all of the spaces: `power = swh_use_value * sum(space.number_of_occupants for space in spaces)`
+  - in this case, swh_use_value is assumed to have units of btu/hr/person, so calculate energy_required by multiplying power (btu/hr) by the sum of the hourly_schedule: `energy_required = power * sum(hourly_schedule)`
+  - return energy_required: `return energy_required`
+- otherwise if use_units is POWER_PER_AREA: `elsif use_units == "POWER_PER_AREA":`
+  - set power equal to the swh_use_value * the sum of the floor area in all of the spaces: `power = swh_use_value * sum(space.floor_area for space in spaces)`
+  - in this case, swh_use_value is assumed to have units of btu/hr/ft2, so calculate energy_required by multiplying power (btu/hr) by the sum of the hourly_schedule: `energy_required = power * sum(hourly_schedule)`
+  - return energy_required: `return energy_required`
+- otherwise if use_units is POWER: `elsif use_units == "POWER":`
+  - in this case, swh_use_value is assumed to have units of btu/hr, so calculate energy_required by multiplying swh_use_value by the sum of the hourly_schedule: `energy_required = swh_use_value * sum(hourly_schedule)`
```

Comment:
Do we agree that it is power across all the spaces as opposed to power per space?

Reply:
I think we should discuss this on the call - this would apply to both POWER and VOLUME.  The general rule that we agreed on a few weeks ago was:

**The rule is that if spaces reference the swh_use, then the use applies to only those spaces.  If the use references no spaces, it applies to all spaces in the building segment.**

However, for VOLUME and POWER, it doesn't make as much sense to me for it to get repeated for each space in a building segment, or for each applied space.

Reply:
Sounds good to discuss on the call

Reply:
discussed on call 7/18 - if it's power or volume, it it's applied to no spaces, only count it once.  If it's explicitly applied to any spaces, then count it once for each space.